configure/os/CONFIG_SITE.Common.RTEMS

```diff
@@ -2,12 +2,27 @@
 #
 # Site-specific information for all RTEMS targets
 #-------------------------------------------------------
+# RTEMS Series and Version
+
+#RTEMS_SERIES = 4.9
```

Comment:
MAD's Request is to not set both these; preferred config to just provide the path to the Makefile.inc file (we include it from `$(RTEMS_BASE)/$(RTEMS_TARGET_CPU)-rtems$(RTEMS_VERSION)/$(RTEMS_BSP)/Makefile.inc`).

HJ says version numbers have changed in the 5 series anyway.

Agree to look at supporting a snapshot of 5 instead of waiting for a stable release.

Reply:
Ok, I'm now using the settings from the Makefile.inc  
! In the Makefile one can find:
“
# This support will be removed from RTEMS. Please consider other
# ways to build applications.
“
The statement of the developers of RTEMS:
There is no agreement among the developers. I think we should remove Makefile.inc because it is problematic to support across all BSPS. Sebastian thinks it is important for existing users and has added support in RTEMS 6?s waf build system. Does this means we support a few popular BSPs or does Sebastian need to determine all will work?

as I am running on a private branch, my changes:

diff --git a/configure/os/CONFIG_SITE.Common.RTEMS b/configure/os/CONFIG_SITE.Common.RTEMS
index 7272d72..7cabdae 100644
--- a/configure/os/CONFIG_SITE.Common.RTEMS
+++ b/configure/os/CONFIG_SITE.Common.RTEMS
@@ -2,16 +2,6 @@
 #
 # Site-specific information for all RTEMS targets
 #-------------------------------------------------------
-# RTEMS Series and Version
-
-#RTEMS_SERIES = 4.9
-#RTEMS_VERSION = 4.9.2
-
-#RTEMS_SERIES = 4.10
-#RTEMS_VERSION = 4.10.2
-
-#RTEMS_SERIES = 5
-#RTEMS_VERSION = 5.1

 # Where to find RTEMS
 #
@@ -20,9 +10,8 @@
 #RTEMS_BASE = /usr/local/vw/rtems/rtems-$(RTEMS_VERSION)

 # FHI:
-RTEMS_SERIES = 5
-RTEMS_VERSION = 5
-RTEMS_BASE = /Users/junkes/MVME2500/RTEMS/rtems-$(RTEMS_VERSION)
+RTEMS_VERSION = 5.0.0-m1912
+RTEMS_BASE = /home/epics/MVME2500/RTEMS/rtems-$(RTEMS_VERSION)

 # Cross-compile toolchain in $(RTEMS_TOOLS)/bin
 #

diff --git a/configure/os/CONFIG.Common.RTEMS b/configure/os/CONFIG.Common.RTEMS
index 247c516..83f3056 100644
--- a/configure/os/CONFIG.Common.RTEMS
+++ b/configure/os/CONFIG.Common.RTEMS
@@ -96,19 +96,17 @@ OS_CLASS = RTEMS

 #--------------------------------------------------
 # operating system API (src/os/<os_class>-<os-api>)
-OS_API_4.7 = kernel
-OS_API_4.8 = kernel
-OS_API_4.9 = kernel
-OS_API_4.10 = kernel
-OS_API_4.11 = $(error RTEMS-4-11 is not currently supported)
-# Later RTEMS versions will use posix, no need to specify
-OS_API = $(firstword $(OS_API_$(RTEMS_SERIES)) posix)
+ifeq ($(RTEMS_HAS_POSIX_API),yes)
+OS_API = posix
+else
+OS_API = kernel
+endif

 #--------------------------------------------------
 # Operating system flags
 #OP_SYS_LDLIBS += -lrtemsCom -lc -lrtemscpu -lCom -lnfs -lm
 OP_SYS_LDLIBS += -lrtemsCom -lrtemscpu -lrtemsbsp
-#ifeq ($(RTEMS_SERIES),5)
+#ifeq ($(RTEMS_API),5)
 OP_SYS_CFLAGS += -D__LINUX_ERRNO_EXTENSIONS__
 OP_SYS_LDLIBS += -ltftpfs -ltelnetd -lbsd -lc
 #endif

Reply:
NB: I just edited Heinz' comment to stop GitHub's Markdown processor from reformatting it. Triple back-ticks ` ``` ` are so useful...

Reply:
II'd cancel that merge request here. I plan to build a completely new request from the current Epics release. For an RTEMS pre-release and Qemu ARM Zynq. Does that make sense?